ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java

```diff
@@ -386,4 +403,25 @@ private void terminateQuery(final TerminateQuery terminateQuery) {
           String.format("No running query with id %s was found", queryId));
     }
   }
+
+  private void throwTooManyActivePersistentQueriesException(final String statementStr) {
+    throw new KsqlException(
+        String.format(
+            "Not executing statement '%s' since the statement causes the limit on number "
```

Comment:
The `ksqlConfig` you're using here isn't the one containing the limit you've just applied. You'll need to pass in the correct config.

Maybe reword the message to:

> Not executing statement 'blah' as it would take the active query limit above the configured limit. Use the TERMINATE command to terminate existing queries, or increase the 'ksql.query.persistent.active.limit' setting via the 'ksql-server.properties' file. currentQueryCount: 6, limit: 6.